sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -779,6 +823,10 @@ public void populateDisplayData(DisplayData.Builder builder) {
             .addIfNotDefault(DisplayData.item("codec", codec.toString())
                 .withLabel("Avro Compression Codec"),
                 DEFAULT_CODEC.toString());
+        for (Map.Entry<String, String> entry : metadata.entrySet()) {
+          builder.add(DisplayData.item(entry.getKey(), entry.getValue()).withLabel("Metadata"));
```

Comment:
This will make all metadata have the same label called "Metadata" irrespective of the key which is not what I think you want.

I think you want to do:
```
builder.include("Metadata", new Metadata());
```

and add the following private class inside Bound:
```
private class Metadata implements HasDisplayData {
  void populateDisplayData(DisplayData.Builder builder) {
    for (Map... entry : ...) {
      String valueString = // convert entry.getValue() to string
      builder.add(DisplayData.item(entry.getKey(), entryString);
    }
  }
}
```

This allows for all the metadata to be grouped under a single namespace component and be logically related.
For the binary data, it won't be human readable and doesn't really fit as information that would be consumed, so I suggest that either:
* skip over specifying binary fields
* add them and make the value a word like "..." to mark that its specified
* add them and make the value the hex or base64 representation of a fixed prefix of the bytes followed by "..." if the bytes is long otherwise the entire hex/base64 representation.
* add them and make the value a checksum of the bytes

@nevillelyh @dhalperi @swegner What kinds of binary data do people add to Avro files and is its presence in the DisplayData genuinely useful?
@nevillelyh @dhalperi @swegner any specific preference for the above?